src/node_url.cc

```diff
@@ -829,6 +829,16 @@ static url_host_type ParseOpaqueHost(url_host* host,
   return type;
 }
 
+static inline bool IsAllASCII(std::string* input) {
```

Comment:
@mscdex : I checked the function, it optimizes for length >=16 . Most domain names will be smaller, so will it help?. We can try and benchmark it.